### ISSUE TITLE
divergence 

### ISSUE THREAD

**Issue by antagomir** [microbiome/mia#64]
Divergence is close to beta diversity as a concept and sometimes used to quantify community heterogeneity. For instance, two different groups of samples may be different in terms of community heterogeneity (divergence). But this has to be balanced with group sizes and against some reference (selected reference sample or community median for instance). The microbiome pkg has one simple implementation for [divergence](https://github.com/microbiome/microbiome/blob/master/R/divergence.R). We could discuss whether there is place for such function in the `mia` pkg.

Can be also used to measure divergence over time but that could be a separate issue to `miaTime`.

**Comment by FelixErnst** [comment]
As far as I understand it, this should do the following:
- calculate a distance measurement select by `method` specifically in a 1:many basis
- return a numeric vector.

Is that correct?

If it is a yes, wouldn't this do the same thing? (mat = abundance matrix, x = the abundance value of a single sample)
```
mat <- cbind(matrix(x,ncol=1),mat)
dist <- calculateDistance(mat)
as.matrix(dist)[,1]
```

**Comment by FelixErnst** [comment]
I think, if we want to implement this functionality, the name has to be more on point.

**Comment by antagomir** [comment]
Yes, this is the point except that there could be additional arguments to tune sample sizes and possible other relevant factors (how the reference sample is chosen etc) because these may influence comparability of the divergence index between different groups. This index is correlated with sample size, for instance and therefore cannot be directly compared between groups with different sample size.

**Comment by FelixErnst** [comment]
In the example given about, the assumption is made, that the reference information are provided as `x`, which can be used to connect to different sample sets. The number of samples wouldn't matter in that case, just the same number of features and by implication the same identity of features would be required.

Is that correct?

**Comment by antagomir** [comment]
Yes, same features should be required so that the comparisons are meaningful.

The function itself could do subsampling for the samples to a predefined sample size (function argument).

**Comment by antagomir** [comment]
This feature is less popular / less widely used and therefore not necessarily top priority.

**Comment by antagomir** [comment]
Merged via #110
